### PR TITLE
Add more interesting state change abilities to the DryRunBMC

### PR DESCRIPTION
**Pull request by jakeschuurmans**
#### What does this PR do

Add ability for the power state of the fake BMC to be remembered, and add boot device to it as well

### CODE REVIEW

File: internal/device/dryRun.go [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
+)
+
+type serverState struct {
+	power      string
+	bootDevice string
+}
+
+var (
+	errBmcCantFindServer = errors.New("dryrun BMC couldnt find server to set state")
+	serverStates         map[string]serverState
```
Comment (DoctorVin) [inline internal/device/dryRun.go:17]: you could initialize this here via `serverStates = map[string]serverState{}` if you wanted and save the nil check below.

**Review by DoctorVin — DISMISSED** [review]

File: internal/device/dryRun.go [hunk 2]
```diff
@@ -25,13 +51,24 @@ func (b *DryRunBMC) Close(_ context.Context) error {
...
+	return "unknown", nil
 }
 
 // SetPowerState sets the given power state on the device
 func (b *DryRunBMC) SetPowerState(_ context.Context, state string) error {
-	b.state = state
-	return nil
+	serverState, ok := serverStates[b.id]
+	if ok {
+		serverState.power = state
```
Comment (zevweiss) [inline internal/device/dryRun.go:62]: Given that the `state` argument can be things that are really more _actions_ (like `"cycle"` or `"reset"`) than actual states, should we maybe do some special-casing of those here?

Comment (jakeschuurmans) [reply internal/device/dryRun.go:62]: Yea could be. Could spin up a go func to change the state after 30 seconds or something if its "reset". IDK. Any suggestions?

Comment (jakeschuurmans) [reply internal/device/dryRun.go:62]: Or a timer to the state, so if "reset" is called we set the timer, and the next time the state is asked for, if its "reset" we check the timer, if 30 seconds have passed, we change the state to "on"

**Review by DoctorVin — APPROVED** [review]

**Pull request merged**